### PR TITLE
using one controler instance per widget

### PR DESCRIPTION
**Pull request by keywan-ghadami-oxid**
widget class does have member variables and by reusing the singleton this get mixed up during nested start calls
so using a new instance allows using nested widget without errors.

### CODE REVIEW

**Comment by Sieg** [comment]
Hello @keywan-ghadami-oxid, do you have any steps to reproduce the errors, or any bug report for this?

**Comment by alfredbez** [comment]
Small hint: I'd reword the commit either to:
```diff
- using one controler instance per widget
+ using one controller instance per widget
```
or
```diff
- using one controler instance per widget
+ using one control instance per widget
```

**Comment by keywan-ghadami-oxid** [comment]
@Sieg it is hard to reproduce. as far I know there is no bug report but the support team was informed about a possible workaround.

to reproduce a error that I had with that you need to create theme with nested widgets that are called within a 404 page. I have no idea how to proceed on this i can only tell that in this case the view chain was somehow mixed up and by that the page close funtion in config object did the wring thing. maybe there are easier way to reproduce but i do not know. anyway i thought better share this solutiin then having this issue to debug again.

**Comment by Sieg** [comment]
Hello @keywan-ghadami-oxid, could you please rebase on the master and reopen the pull request?

**Comment by michaelkeiluweit** [comment]
The case is reproducible when having nested widgets and by provoking the 404 page.

Example:
Open: _source/Application/views/flow/tpl/widget/header/categorylist.tpl_
Search for: 
```<div class="[{if $blFullwidth}]container[{else}]container-fluid[{/if}]">```

Replace it with:
```
<div class="[{if $blFullwidth}]container[{else}]container-fluid[{/if}]">
[{oxid_include_widget cl="oxwLanguageList" lang=$oViewConf->getActLanguageId() _parent=$oView->getClassName() nocookie=1 _navurlparams=$oViewConf->getNavUrlParams() anid=$oViewConf->getActArticleId()}]
```


Call a page which doesn't exist. For example: http://localhost/en/Kiteboarding/Kites/Kite-CO-GT.html

Normally you should see the 404 page but you willl be redirected because an exception was thrown:    
```
[2020-07-17 11:05:35] OXID Logger.ERROR: Call to a member function pageClose() on null ["[object] (Error(code: 0): Call to a member function pageClose() on null at /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/Config.php:660)\n[stacktrace]\n#0 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/ShopControl.php(305): OxidEsales\\EshopCommunity\\Core\\Config->pageClose()\n#1 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/ShopControl.php(142): OxidEsales\\EshopCommunity\\Core\\ShopControl->_process('OxidEsales\\\\Esho...', NULL, Array, NULL)\n#2 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/WidgetControl.php(65): OxidEsales\\EshopCommunity\\Core\\ShopControl->start('oxwcategorytree', NULL, Array, NULL)\n#3 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/Smarty/Plugin/function.oxid_include_widget.php(32): OxidEsales\\EshopCommunity\\Core\\WidgetControl->start('oxwcategorytree', NULL, Array, NULL)\n#4 /var/www/html/source/tmp/smarty/fab621f22b86deb811bccc5b2a34acac^%%93^93D^93D6031B%%header.tpl.php(84): smarty_function_oxid_include_widget(Array, Object(Smarty))\n#5 /var/www/html/vendor/smarty/smarty/libs/Smarty.class.php(1876): include('/var/www/html/s...')\n#6 /var/www/html/source/tmp/smarty/fab621f22b86deb811bccc5b2a34acac^%%36^366^366ECF91%%page.tpl.php(14): Smarty->_smarty_include(Array)\n#7 /var/www/html/vendor/smarty/smarty/libs/Smarty.class.php(1876): include('/var/www/html/s...')\n#8 /var/www/html/source/tmp/smarty/fab621f22b86deb811bccc5b2a34acac^%%DA^DA0^DA07DFA9%%err_404.tpl.php(56): Smarty->_smarty_include(Array)\n#9 /var/www/html/vendor/smarty/smarty/libs/Smarty.class.php(1270): include('/var/www/html/s...')\n#10 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Internal/Framework/Smarty/Legacy/LegacySmartyEngine.php(68): Smarty->fetch('message/err_404...')\n#11 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Internal/Framework/Templating/TemplateRenderer.php(35): OxidEsales\\EshopCommunity\\Internal\\Framework\\Smarty\\Legacy\\LegacySmartyEngine->render('message/err_404...', Array)\n#12 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/UtilsView.php(126): OxidEsales\\EshopCommunity\\Internal\\Framework\\Templating\\TemplateRenderer->renderTemplate('message/err_404...', Array)\n#13 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/Utils.php(1463): OxidEsales\\EshopCommunity\\Core\\UtilsView->getTemplateOutput('message/err_404...', Object(OxidEsales\\Eshop\\Application\\Controller\\FrontendController))\n#14 /var/www/html/source/overridablefunctions.php(61): OxidEsales\\EshopCommunity\\Core\\Utils->handlePageNotFoundError('en/Kiteboarding...')\n#15 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/SeoDecoder.php(237): error_404_handler('en/Kiteboarding...')\n#16 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/Config.php(370): OxidEsales\\EshopCommunity\\Core\\SeoDecoder->processSeoCall()\n#17 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/Config.php(470): OxidEsales\\EshopCommunity\\Core\\Config->_processSeoCall()\n#18 /var/www/html/vendor/oxid-esales/oxideshop-ee/Core/Config.php(563): OxidEsales\\EshopCommunity\\Core\\Config->initializeShop()\n#19 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/Config.php(439): OxidEsales\\EshopEnterprise\\Core\\Config->initializeShop()\n#20 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/ShopControl.php(581): OxidEsales\\EshopCommunity\\Core\\Config->init()\n#21 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/ShopControl.php(136): OxidEsales\\EshopCommunity\\Core\\ShopControl->_runOnce()\n#22 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/Oxid.php(27): OxidEsales\\EshopCommunity\\Core\\ShopControl->start()\n#23 /var/www/html/source/index.php(19): OxidEsales\\EshopCommunity\\Core\\Oxid::run()\n#24 /var/www/html/source/oxseo.php(29): require('/var/www/html/s...')\n#25 {main}\n"] []
[17 Jul 11:05:35.034039 2020] [uncaught error] [type E_ERROR] [file /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/Config.php] [line 660] [code ] [message Uncaught Error: Call to a member function pageClose() on null in /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/Config.php:660
Stack trace:
#0 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/ShopControl.php(305): OxidEsales\EshopCommunity\Core\Config->pageClose()
#1 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/ShopControl.php(142): OxidEsales\EshopCommunity\Core\ShopControl->_process('OxidEsales\\Esho...', NULL, Array, NULL)
#2 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/WidgetControl.php(65): OxidEsales\EshopCommunity\Core\ShopControl->start('oxwcategorytree', NULL, Array, NULL)
#3 /var/www/html/vendor/oxid-esales/oxideshop-ce/source/Core/Smarty/Plugin/function.oxid_include_widget.php(32): OxidEsales\EshopCommunity\Core\WidgetControl->start('oxwcategorytree', NULL, Array, NULL)
#4 /var/www/html/source/tmp/smarty/fab621f22b86deb811bccc5b2a34acac^%%93^93D^93D6031B%%header.tpl.php(84): smarty_function_oxid_include_widget(Array, Object(Smarty))
#5 /var/www/html/ven]
```

=> Calling existing pages is working fine. So only the 404 page does cause the error.   

  
[Changing the file as suggested by this PR](https://github.com/OXID-eSales/oxideshop_ce/pull/757/files) does the trick. Also the provided module from Keywan to the Support works (#1168522).
